paddle/fluid/pybind/tensor_py.h

```diff
@@ -510,9 +511,26 @@ inline py::array TensorToPyArray(const framework::Tensor &tensor) {
   std::string py_dtype_str = details::TensorDTypeToPyDTypeStr(tensor.type());
 
   if (!is_gpu_tensor) {
-    return py::array(py::buffer_info(
-        const_cast<void *>(tensor_buf_ptr), sizeof_dtype, py_dtype_str,
-        static_cast<size_t>(tensor.dims().size()), py_dims, py_strides));
+    if (!need_deep_copy) {
```

Comment:
Why use `if (need_deep_copy)` to guard the following code? If the tensor is located on CPU, we do not need to copy. `numpy.array()` would deep copy data by default. This change would make the 2 copies when calling `numpy.array(tensor)` in Python.

Reply:
Discussed online. By default, `need_deep_copy` is False. Thanks a lot.